src/prefect/_internal/concurrency/executor.py

```diff
@@ -43,28 +36,22 @@ class Executor(BaseExecutor):
     def __init__(
         self,
         worker_type: WorkerType = "thread",
-        max_workers: Optional[int] = None,
+        **worker_pool_kwargs: Any,
```

Comment:
I'm guessing this is to accommodate the different workers, which seems reasonable enough, but `max_workers` is going to be universal across executors right? It feels like it might be nice to have it documented as an explicit parameter.

Reply:
The only reason I take "thread"/"process" explicitly (instead of a `BaseExecutor` type) is because we need to know if context vars can be copied over successfully. `BaseExecutor` does not define any `__init__` arguments so for future compatibility with other executor types I'm hesitant to define the parameters here.